filament/backend/include/private/backend/HandleAllocator.h

```diff
@@ -201,6 +205,32 @@ class HandleAllocator {
         return handle_cast<Dp>(const_cast<Handle<B>&>(handle));
     }
 
+    void associate_handle(HandleBase::HandleId id, std::string&& tag) noexcept {
+        // TODO: for now, only pool handles check for use-after-free, so we only keep tags for
+        // those
+        if (isPoolHandle(id)) {
+            // Truncate the tag's age to N bits.
+            constexpr uint8_t N = 2; // support a history of 4 tags
```

Comment:
so that's just so we don't use too much memory per handle? i.e. we keep 4 instead of 16?

Reply:
Yeah exactly. Not sure if we want to make this configurable by the user.